### PR TITLE
docs: update Python migration guide for drop-callback change

### PR DESCRIPTION
**Pull request by wyfo**

### CODE REVIEW

File: content/docs/migration_1.0/Python.md [hunk 1]
```diff
@@ -41,6 +41,30 @@ with zenoh.open(zenoh.Config()) as session:
     # and it will undeclare the subscriber
 ```
 
+## Drop-callback has to be wrapped in `handlers.Callback`
+
+In previous version, it was possible to pass a drop-callback with the main callback in a tuple for operations like `Session.declare_subscriber`. However, it was also possible to pass a tuple with a "receiver" (renamed "handler" in 1.0) as second member, and that could be confusing
```
Comment (Mallets) [inline content/docs/migration_1.0/Python.md:46]:
```suggestion
In the previous 0.11.0 version, it was possible to pass a drop-callback with the main callback in a tuple for operations like `Session.declare_subscriber`. However, it was also possible to pass a tuple with a "receiver" (renamed "handler" in 1.0.0) as second member, and that could lead to confuse for users.
```

File: content/docs/migration_1.0/Python.md [hunk 2]
```diff
@@ -41,6 +41,30 @@ with zenoh.open(zenoh.Config()) as session:
     # and it will undeclare the subscriber
 ```
 
+## Drop-callback has to be wrapped in `handlers.Callback`
+
+In the previous 0.11.0 version, it was possible to pass a drop-callback with the main callback in a tuple for operations like `Session.declare_subscriber`. However, it was also possible to pass a tuple with a "receiver" (renamed "handler" in 1.0.0) as second member, and that could lead to confuse for users.
```
Comment (oteffahi) [inline content/docs/migration_1.0/Python.md:46]:
```suggestion
In the previous 0.11.0 version, it was possible to pass a drop-callback with the main callback in a tuple for operations like `Session.declare_subscriber`. However, it was also possible to pass a tuple with a "receiver" (renamed "handler" in 1.0.0) as second member, and that could lead to confusion for users.
```

File: content/docs/migration_1.0/Python.md [hunk 3]
```diff
@@ -41,6 +41,30 @@ with zenoh.open(zenoh.Config()) as session:
...
+
+- Zenoh 1.0.0
+
+```python
+def on_sample(sample: zenoh.Sample): ...
+def on_done(): ...
+session.declare_subscriber(zenoh.handlers.Callback(on_sample, on_done))
+```
+
+NOTE: ⚠️ Passing drop-callback in a tuple will no more work as expected, as the drop callback will never be executed. To ease migration and avoid surprises, a warning will be displayed in this case.
```
Comment (oteffahi) [inline content/docs/migration_1.0/Python.md:66]:
```suggestion
NOTE: ⚠️ Passing drop-callback in a tuple will no longer work as expected, as the drop callback will never be executed. To ease migration and avoid surprises, a warning will be displayed in this case.
```

**Review by oteffahi — CHANGES_REQUESTED** [review]

File: content/docs/migration_1.0/Python.md (hunk 2 above)
Comment (fuzzypixelz) [reply content/docs/migration_1.0/Python.md]: > and that could confuse users.

The audience is the user and so the user should be addressed directly.

Comment (oteffahi) [reply content/docs/migration_1.0/Python.md]: My suggestion was follow-up to Luca's suggestion which wasn't applied correctly in the previous commit.

**Pull request merged**